### PR TITLE
Add read_raw wrapper function

### PR DESCRIPTION
**Pull request by cbrnr**
Would it be useful to have a wrapper `read_raw` function that calls specific readers based on the file extensions?

### CODE REVIEW

**Comment by christianbrodbeck** [comment]
I think an issue with this was that different readers would need different parameters, for example KIT raw files need specification for the stim-channel that other formats don’t need…

**Comment by larsoner** [comment]
See also discussion in #1930

**Comment by cbrnr** [comment]
I'd argue not to replace the existing `read_raw_*` functions with a generic `read_raw` function. In fact, `read_raw` should just be a wrapper to make it easier for users to load common data formats.

@christianbrodbeck `read_raw` only needs one argument (the file name), I'm passing everything else to the corresponding `read_raw_*` functions. So this shouldn't be an issue.

**Comment by jasmainak** [comment]
@teonbrooks already implemented something similar in our [mne-bids](https://github.com/mne-tools/mne-bids) repo.

File: mne/io/raw.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+from .ctf import read_raw_ctf
+from .edf import read_raw_edf
+from .egi import read_raw_egi
+from .kit import read_raw_kit
+from .fiff import read_raw_fif
+from .nicolet import read_raw_nicolet
+from .eeglab import read_raw_eeglab
+
+
+def read_raw(input_fname, **kwargs):
```
Comment (agramfort) [inline mne/io/raw.py]: then how do you find out what parameters to pass? I find it clearer to be explicit by using the proper dedicated function

**Comment by teonbrooks** [comment]
I think that there may be some value in exploring a generic `read_raw` function, e.g. support for bids export as @jasmainak mentioned above, but I do think we should work out standardizing the standards across the different file types instead of the catchall `**kwargs`. 
the problem with the `kwargs` is that you often need to reference the format specifics if all the formats aren't listed in the docstring and at that point, you could just use the existing function. I think if we did have it, we would want it to cover of ~80% use case, and then point to specific readers for add'l functionality.

just my 2c

**Comment by jasmainak** [comment]
> standardizing the standards

The irony! :)

**Comment by cbrnr** [comment]
Yes, I agree. If we just pass ``**kwargs`` the user has to read the doc of the specific function, so this doesn't really make things easier. I just thought it is kind of silly from a user's perspective to have multiple reader functions if the goal is to read an EEG/MEG raw file - most users won't care about the format, they just want to load the data. So yeah, either we start discussing a complete revamp of the `io` functions, or we'll just leave it as is for now.

**Comment by jona-sassenhagen** [comment]
I think I agree with @teonbrooks  - if the args of the different readers are 95% the same, so that a generic wrapper would have 95% of the docstring for the specific functions, we could probably have a tiny bit of **kwargs at the end.

We are, after all, doing this with montages already.

File: mne/io/raw.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    elif ext == '.mff':
+        return read_raw_egi(input_fname, **kwargs)
+    elif ext == '.sqd':
+        return read_raw_kit(input_fname, **kwargs)
+    elif ext == '.fif':
+        return read_raw_fif(input_fname, **kwargs)
+    elif ext == '.data':
+        return read_raw_nicolet(input_fname, **kwargs)
+    elif ext == '.set':
+        return read_raw_eeglab(input_fname, **kwargs)
```
Comment (jona-sassenhagen) [inline mne/io/raw.py]: In principle, I'd use a dict for this.

**Comment by agramfort** [comment]
I am still -1

it means testing twice many IO things. But I would be ok to have a private
helper if you need it in your GUI code.

**Comment by cbrnr** [comment]
I agree, this is probably not such a great idea. The GUI will abstract this away anyway, so there's no need for a (private) helper.

**Pull request closed without merging**